api/dbadapters/index.ts

```diff
@@ -38,7 +38,8 @@ export interface IDbAdapter extends IDbClient {
     projectConfig?: dataform.IProjectConfig
   ): Promise<dataform.IQueryEvaluation[]>;
 
-  schemas(database: string): Promise<string[]>;
+  databases(): Promise<string[]>;
+  schemas(databases?: string | string[]): Promise<string[]>;
```

Comment:
do we need the overload? can we just always require an array, for simplicity's sake?